app/controllers/api/base_controller/renderer.rb

```diff
@@ -367,6 +367,7 @@ def validate_attr_selection(resource)
         physical_attrs, virtual_attrs = [], []
         attrs = attribute_selection
         return [physical_attrs, virtual_attrs] if resource.kind_of?(Hash) || attrs == 'all'
+        return [attrs, virtual_attrs] if (attrs - ID_ATTRS).empty?
```

Comment:
Does removing the condition in the ternary on 382/3 (and defaulting to the RHS) have the same effect? I'm wondering if that would be preferable - if you could fix this by only removing code 😁 ✂️

Reply:
@imtayadeway I tried that 😢  it was the 🐰 hole I started to go down, actually. I can follow up with refactoring, though